src/angularComponent/ngcOmniboxSuggestionsDirective.js

```diff
@@ -97,3 +20,62 @@ export default function ngcOmniboxSuggestionsDirective($document, $templateCache
     }
   };
 }
+
+/**
+ * Takes in the public API template submitted by the app-maker and modifies it to repeat through
+ * the data structure of the Omnibox. The template will be different depending on whether it finds
+ * an ngc-omnibox-suggestion-category or not.
+ *
+ * @private
+ * @param {HTMLElement} element -- HTMLElement that contains the template to modify
+ * @param {jQLiteElement} $document
+ * @param {Object} $templateCache
+ * @param {String} templateCacheName -- Unique string to use as the template cache name for
+ *     recursively rendering the template
+ * @returns {String} Modified template string
+ */
+export function getModifiedTemplate(element, $document, $templateCache,
+    templateCacheName = `category-tmpl-${new Date().getTime() * Math.random()}`) {
```

Comment:
When I'm testing, otherwise I have no way of enforcing the values are the same for the input and output and the test will fail.